pkgs/development/python-modules/pymongo-inmemory/default.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, fetchpatch
+, poetry-core
+, pymongo
+, pytestCheckHook
+, click
+}:
+
+buildPythonPackage rec {
+  pname = "pymongo-inmemory";
+  version = "0.3.0";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "kaizendorks";
+    repo = "pymongo_inmemory";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-2ShI6QgHEOv57rPAJjRpgS+8LjEY9XdMEpqMdpEszM8=";
+  };
+
+  # fix "ModuleNotFoundError: No module named 'poetry.masonry'"
+  patches = [
+    (fetchpatch { # https://github.com/kaizendorks/pymongo_inmemory/pull/85
+      url = "https://github.com/kaizendorks/pymongo_inmemory/commit/58c995b67a7812ecebc9acfc596df59067b2abad.patch";
+      hash = "sha256-Xvs7kZBGtZwHz0VvEU200rAmknVy8QSaDgVaQ2Q1Dz0=";
+    })
+  ];
+
+  postPatch = ''
+    # move cache location from nix store to home
+    substituteInPlace pymongo_inmemory/context.py \
+      --replace \
+        'CACHE_FOLDER = path.join(path.dirname(__file__), "..", ".cache")' \
+        'CACHE_FOLDER = os.environ.get("XDG_CACHE_HOME", os.environ["HOME"] + "/.cache") + "/pymongo-inmemory"'
+
+    # fix a broken assumption arising from the above fix
+    substituteInPlace pymongo_inmemory/_utils.py \
+      --replace \
+        'os.mkdir(current_path)' \
+        'os.makedirs(current_path)'
+  '';
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    pymongo
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    click
```

Comment:
Shouldn't this be in `propagatedBuildInputs`?

Reply:
https://github.com/kaizendorks/pymongo_inmemory/blob/master/pyproject.toml#L44

Reply:
Yes, but look at where it's imported. Maybe upstream misdeclared their dependencies?

Reply:
Good catch! :smiley:

In my own projects I've included click because i use it dev utility scripts, so i didn't think twice about it.

Reply:
It is indeed an internal tool :S https://github.com/kaizendorks/pymongo_inmemory/issues/86#issuecomment-1712916872

Reply:
Then it should be in neither `nativeBuildInputs` nor `propagatedBuildInputs`.